$NetBSD$
the second part of this patch is from
LFS and http://www.geocrawler.com/mail/msg.php3?msg_id=6612801&list=709

--- syslogd.c.orig	2001-03-11 11:40:10.000000000 -0800
+++ syslogd.c
@@ -521,7 +521,8 @@ static char sccsid[] = "@(#)syslogd.c	5.
 #if defined(FSSTND)
 #define _PATH_LOGPID _PATH_VARRUN SYSLOGD_PIDNAME
 #else
-#define _PATH_LOGPID "/etc/" SYSLOGD_PIDNAME
+/* #define _PATH_LOGPID "/etc/" SYSLOGD_PIDNAME */
+#define _PATH_LOGPID _PATH_VARRUN SYSLOGD_PIDNAME
 #endif
 #else
 #ifndef _PATH_LOGPID
@@ -890,11 +891,11 @@ int main(argc, argv)
 		dprintf("Checking pidfile.\n");
 		if (!check_pid(PidFile))
 		{
+			signal (SIGTERM, doexit);
 			if (fork()) {
 				/*
 				 * Parent process
 				 */
-				signal (SIGTERM, doexit);
 				sleep(300);
 				/*
 				 * Not reached unless something major went wrong.  5
